base/compiler/abstractinterpretation.jl

```diff
@@ -789,6 +788,21 @@ function abstract_eval_cfunction(e::Expr, vtypes::VarTable, sv::InferenceState)
     nothing
 end
 
+# convert an inferred static parameter value to the inferred type of a static_parameter expression
+function sparam_type(@nospecialize(val))
+    if isa(val, TypeVar)
+        if Any <: val.ub
```

Comment:
`val.ub` may be a TypeVar

Reply:
This PR only moved this code. But also I think vars in bounds are normalized away in inferencestate.jl.

Reply:
Ah, OK – that makes sense. I _was_ wondering what all of those loops were doing there.